File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/messaging/MessagingExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports com.db4o
Imports com.db4o.messaging


Namespace com.db4odoc.f1.messaging

    Public Class MessagingExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub ConfigureServer()
            Dim objectServer As ObjectServer = Db4o.OpenServer(YapFileName, 0)
            objectServer.Ext().Configure().SetMessageRecipient(New SimpleMessageRecipient())
            Try
                Dim clientObjectContainer As ObjectContainer = objectServer.OpenClient()
                ' Here is what we would do on the client to send the message
                Dim sender As MessageSender = clientObjectContainer.Ext().Configure()
                sender.Send(New MyClientServerMessage("Hello from client."))
                clientObjectContainer.Close()
            Finally
                objectServer.Close()
            End Try
        End Sub
        ' end ConfigureServer
    End Class

    Public Class SimpleMessageRecipient
        Implements MessageRecipient
        Public Sub ProcessMessage(ByVal objectContainer As ObjectContainer, ByVal message As Object) Implements MessageRecipient.ProcessMessage
            ' message objects will arrive in this code block
            System.Console.WriteLine(message)
        End Sub
    End Class

End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/messaging/MessagingExample.cs
using System;
using com.db4o;
using com.db4o.messaging;


namespace com.db4odoc.f1.messaging
{

	public class MessagingExample 
	{
		public readonly static string YapFileName = "formula1.yap";
	
		public static void ConfigureServer() 
		{
			ObjectServer objectServer = Db4o.OpenServer(YapFileName, 0);
			objectServer.Ext().Configure().SetMessageRecipient(new SimpleMessageRecipient());
			try 
			{
				ObjectContainer clientObjectContainer = objectServer.OpenClient();
				// Here is what we would do on the client to send the message
				MessageSender sender = clientObjectContainer.Ext().Configure()
					.GetMessageSender();

				sender.Send(new MyClientServerMessage("Hello from client."));
				clientObjectContainer.Close();
			} 
			finally 
			{
				objectServer.Close();
			}
		}
		// end ConfigureServer 
	}

	public class SimpleMessageRecipient: MessageRecipient
	{
		public void ProcessMessage(ObjectContainer objectContainer,Object message) 
		{
			// message objects will arrive in this code block
			System.Console.WriteLine(message);
		}
	}
}
